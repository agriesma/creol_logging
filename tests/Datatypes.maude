***
*** Datatypes.maude -- Test suite for the data types module.
***
*** Copyright (c) 2007
***
*** The purpose of this file is to create the files `prelude.creol'
*** and `modelcheck.creol'.  These files have to be processed with
*** m4, with either one of `CREOL' or `MODELCHECK' defined.
***
*** This program is free software; you can redistribute it and/or
*** modify it under the terms of the GNU General Public License as
*** published by the Free Software Foundation; either version 2 of the
*** License, or (at your option) any later version.
***
*** This program is distributed in the hope that it will be useful, but
*** WITHOUT ANY WARRANTY; without even the implied warranty of
*** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
*** General Public License for more details.
***
*** You should have received a copy of the GNU General Public License
*** along with this program; if not, write to the Free Software
*** Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
*** 02111-1307, USA.
***

set show timing off .
set show advise off .
load creol-datatypes

red in CREOL-DATA-SIG : emp .
red in CREOL-DATA-SIG : emp :: emp .
red in CREOL-DATA-VIDLIST : noVid .
red in CREOL-DATA-VIDLIST : noVid , noVid .
red in CREOL-DATA-SIG : "a" .
red in CREOL-DATA-SIG : "a" :: "a" .
red in CREOL-DATA-VIDLIST : "a" , "a" .
red in CREOL-EXPRESSION : "a" ( "a" ) .
red in CREOL-DATA-SIG : null .
red in CREOL-DATA-SIG : null :: null .
red in CREOL-DATA-SIG : null :: "a" .
red in CREOL-EXPRESSION : null :: ("a" ( "a" )) :: null .
red in CREOL-EXPRESSION : ("a" ( "a" )) :: null .
red in CREOL-DATA-VIDLIST : null , "a" .
red in CREOL-DATA-BOOL : bool(true) .
red in CREOL-DATA-BOOL : "=" ( null :: null ) .
red in CREOL-DATA-BOOL : "/=" ( null :: null ) .
red in CREOL-DATA-BOOL : "~" ( "=" ( null :: null ) ) .
red in CREOL-DATA-BOOL : "||" ( ("=" ( null :: null )) :: ("=" ( null :: null )) ) .
red in CREOL-DATA-LIST : "|-" (list(int(1) :: int(2)) :: "head" (list(int(1) :: int(2))) ) .
red in CREOL-DATA-LIST : "#" (list(emp)) .
red in CREOL-DATA-LIST : "#" (list(int(1) :: int(2))) .

red in CREOL-DATA-SET : "add" (set(emptyset) :: int(0)) .
red in CREOL-DATA-SET : "add" (set(int(0)) :: int(0)) .
red in CREOL-DATA-SET : "add" (set(int(1)) :: int(0)) .

red in CREOL-DATA-SET : "remove" (set(emptyset) :: int(0)) .
red in CREOL-DATA-SET : "remove" (set(int(0)) :: int(0)) .
red in CREOL-DATA-SET : "remove" (set(int(1)) :: int(0)) .

red in CREOL-DATA-MAP : "in" (int(0) :: map(empty)) .
red in CREOL-DATA-MAP : "insert" (map(empty) :: int(0) :: int(0)) .

quit
