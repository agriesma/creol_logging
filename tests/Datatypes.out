==========================================
reduce in CREOL-DATA-SIG : emp .
result DataList: emp
==========================================
reduce in CREOL-DATA-SIG : emp .
result DataList: emp
==========================================
reduce in CREOL-DATA-VIDLIST : noVid .
result VidList: noVid
==========================================
reduce in CREOL-DATA-VIDLIST : noVid .
result VidList: noVid
==========================================
reduce in CREOL-DATA-SIG : "a" .
result Char: "a"
==========================================
reduce in CREOL-DATA-SIG : "a" :: "a" .
result NeExprList: "a" :: "a"
==========================================
reduce in CREOL-DATA-VIDLIST : "a", "a" .
result NeVidList: "a", "a"
==========================================
reduce in CREOL-EXPRESSION : "a"("a") .
result Expr: "a"("a")
==========================================
reduce in CREOL-DATA-SIG : null .
result Data: null
==========================================
reduce in CREOL-DATA-SIG : null :: null .
result NeDataList: null :: null
==========================================
reduce in CREOL-DATA-SIG : null :: "a" .
result NeExprList: null :: "a"
==========================================
reduce in CREOL-EXPRESSION : null :: "a"("a") :: null .
result NeExprList: null :: "a"("a") :: null
==========================================
reduce in CREOL-EXPRESSION : "a"("a") :: null .
result NeExprList: "a"("a") :: null
==========================================
reduce in CREOL-DATA-VIDLIST : null, "a" .
result [ExprList,VidList]: null, "a"
==========================================
reduce in CREOL-DATA-BOOL : bool(true) .
result Data: bool(true)
==========================================
reduce in CREOL-DATA-BOOL : "="(null :: null) .
result Data: bool(true)
==========================================
reduce in CREOL-DATA-BOOL : "/="(null :: null) .
result Data: bool(false)
==========================================
reduce in CREOL-DATA-BOOL : "~"("="(null :: null)) .
result Data: bool(false)
==========================================
reduce in CREOL-DATA-BOOL : "||"("="(null :: null) :: "="(null :: null)) .
result Data: bool(true)
==========================================
reduce in CREOL-DATA-LIST : "|-"(list(int(1) :: int(2)) :: "head"(list(int(1)
    :: int(2)))) .
result Data: list(int(1) :: int(2) :: int(1))
==========================================
reduce in CREOL-DATA-LIST : "#"(list(emp)) .
result Data: int(0)
==========================================
reduce in CREOL-DATA-LIST : "#"(list(int(1) :: int(2))) .
result Data: int(2)
==========================================
reduce in CREOL-DATA-SET : "add"(set(emptyset) :: int(0)) .
result Data: set(int(0))
==========================================
reduce in CREOL-DATA-SET : "add"(set(int(0)) :: int(0)) .
result Data: set(int(0))
==========================================
reduce in CREOL-DATA-SET : "add"(set(int(1)) :: int(0)) .
result Data: set(int(0) : int(1))
==========================================
reduce in CREOL-DATA-SET : "remove"(set(emptyset) :: int(0)) .
result Data: set(emptyset)
==========================================
reduce in CREOL-DATA-SET : "remove"(set(int(0)) :: int(0)) .
result Data: set(emptyset)
==========================================
reduce in CREOL-DATA-SET : "remove"(set(int(1)) :: int(0)) .
result Data: set(int(1))
==========================================
reduce in CREOL-DATA-MAP : "in"(int(0) :: map(empty)) .
result Data: bool(false)
==========================================
reduce in CREOL-DATA-MAP : "insert"(map(empty) :: int(0) :: int(0)) .
result Data: map(mapentry(int(0), int(0)))
Bye.
